packages/backend-data/src/factory.ts

```diff
@@ -149,7 +156,43 @@ class DataGenerator implements ConstructContainerEntryGenerator {
         ? this.props.schema.schemas
         : [this.props.schema];
 
-      schemas.forEach((schema) => {
+      const isSandboxDeployment =
+        scope.node.tryGetContext(CDKContextKey.DEPLOYMENT_TYPE) === 'sandbox';
+
+      // get the branch name and use the imported table map for that key
+      // use the sandbox key when in sandbox deployment
+      const amplifyBranchName = isSandboxDeployment
+        ? 'sandbox'
+        : scope.node.tryGetContext(CDKContextKey.BACKEND_NAME);
+      // ensure all branch names are unique
+      if (this.props.migratedAmplifyGen1DynamoDbTableMappings) {
+        // Remove this warning for GA
+        Annotations.of(scope).addWarning(
+          'migratedAmplifyGen1DynamoDbTableMappings is experimental and is not recommended for production use. This functionality may be changed or removed without warning.',
+        );
```

Comment:
How does this warning work? Looking at the cdk docs, it seems the `addWarning` adds a un-acknowledgeable warning and fails in the `--strict` mode. Both sandbox and pipeline deployments are run in the `strict` mode.

Using `addWarningV2` also would fail as per the docs https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.Annotations.html#addwbrwarningwbrv2id-message

Reply:
The data construct is already using `Annotations.addWarning` and we haven't seen the deployments fail in strict mode. https://github.com/aws-amplify/amplify-category-api/blob/d6815d5fff632c648120e9cdbfe94beb696e7e44/packages/amplify-graphql-api-construct/src/amplify-graphql-api.ts#L181

Maybe the docs are inaccurate? I'll try testing this again.

Reply:
This might be a bug in CDK if docs are correct such that if they fix we would fail. We should discuss this with CDK team.

Reply:
I confirmed that is deploying fine with `addWarning`. You are right that we could get the rug pulled though. I think we can change this to use `addInfo` for now. This will give us the same functionality of printing the message at synth time.

Then we can follow up with CDK to see why warning isn't working as expected.